backends/p4tools/common/lib/coverage.cpp

```diff
@@ -0,0 +1,59 @@
+#include "backends/p4tools/common/lib/coverage.h"
+
+#include <ostream>
+
+#include "lib/log.h"
+
+namespace P4Tools {
+
+namespace Coverage {
+
+CollectStatements::CollectStatements(CoverageSet& output) : statements(output) {}
+
+bool CollectStatements::preorder(const IR::AssignmentStatement* stmt) {
```

Comment:
can't this be done for the Statement class?

Reply:
I believe this was done because not all statements are relevant. For example, switch or block statements are only wrappers for the assignment and method call statements contained within.